src/js/core/methods/EthereumSignTypedData.js

```diff
@@ -36,21 +42,54 @@ export default class EthereumSignTypedData extends AbstractMethod<'ethereumSignT
 
         this.info = getNetworkLabel('Sign #NETWORK typed data', network);
 
-        const { data, metamask_v4_compat } = payload;
+        // const { data, metamask_v4_compat } = payload;
 
-        this.params = {
-            path,
-            data,
-            metamask_v4_compat,
-        };
+        this.params = { ...payload, address_n: path };
     }
 
     async run() {
         const cmd = this.device.getCommands();
-        const { path: address_n, data, metamask_v4_compat } = this.params;
+        const { address_n, metamask_v4_compat } = this.params;
+
+        if (this.device.features.model === '1') {
+            if (!this.params.metamask_v4_compat) {
```

Comment:
This should not be here, only in the plugin. T1 will happily sign hashes generated as-if with `metamask_v4_compat=false` -- it's the plugin that is unable to _generate_ this data.

Reply:
thanks, removed.